File: routine-implementations/agentconf/client/src/main/java/at/enfilo/def/agentconf/client/util/CSVReader.java
package at.enfilo.def.agentconf.client.util;

import java.io.*;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.LinkedList;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class CSVReader {

    public static List<String> readLinesFromFile(Path filePath) {

        BufferedReader reader = null;
        List<String> lines = new LinkedList<>();

        try {
            //URL resource = CSVReader.class.getClassLoader().getResource(filePath);
            File file = new File(filePath.toString());
            FileInputStream fis = new FileInputStream(file);
            reader = new BufferedReader(new InputStreamReader(fis));
            String line = null;

            while ((line = reader.readLine()) != null) {
                if (!line.toLowerCase().contains("milliseconds")) {
                    lines.add(line);
                }
            }
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
          e.printStackTrace();
        } finally {
            if (reader != null) {
                try {
                    reader.close();
                } catch (IOException e) {
                    e.printStackTrace();;
                }
            }
        }
        return lines;
    }

    public static List<Path> getAllFilesFromFolder(String folderPath, String pattern) {
        try (Stream<Path> paths = Files.walk(Paths.get(folderPath))) {
            return paths.filter(Files::isRegularFile)
                .filter(p -> {
                    if (pattern != null && !pattern.isEmpty()) {
                        return p.toString().matches(pattern);
                    }
                    if (p.getFileName().toString().startsWith(".")) {
                        return false;
                    }
                    return true;
                })
                .collect(Collectors.toList());
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: client-implementations/agentconf-csharp-client/util/CSVReader.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace agentconf_csharp_client.util
{
    public static class CSVReader
    {
        public static List<string> ReadLinesFromFile(string filePath)
        {
            List<string> lines = new List<string>();

            using (var fileStream = File.OpenRead(filePath))
            {
                using (var streamReader = new StreamReader(fileStream, Encoding.UTF8))
                {
                    String line;
                    while((line = streamReader.ReadLine()) != null)
                    {
                        if (!line.ToLower().Contains("milliseconds"))
                        {
                            lines.Add(line);
                        }
                    }
                }
            }
            return lines;
        }

        public static List<string> GetAllFilesFromFolder(string folderPath, string pattern)
        {
            DirectoryInfo directory = new DirectoryInfo(folderPath);
            FileInfo[] files = null;
            if (String.IsNullOrEmpty(pattern))
            {
                files = directory.GetFiles();
            }
            else
            {
                files = directory.GetFiles(pattern);
            }

            List<string> filePaths = new List<string>();
            foreach (FileInfo file in files)
            {
                if (!file.Name.StartsWith(".", StringComparison.InvariantCultureIgnoreCase))
                {
                    filePaths.Add(file.Name);
                }
            }
            return filePaths;
        }
    }
}
